### ISSUE TITLE
Asset types don't seem to get resolved on all switch statement cases

### ISSUE THREAD

**Issue by colinator27** [krzys-h/UndertaleModTool#59]
This may not be limited to switch statements, but is a little bit of an issue.

An example of this is in Deltarune's `__background_set_internal` script:
```js
switch local.__prop
    {
        case e__BG.Visible:
            {
                layer_set_visible(local.__layerid, local.__val)
            }
            break
        
        case 2:
            {
                layer_background_change(local.__backid, local.__val)
            }
            break
        
        case 3:
            {
                layer_x(local.__layerid, local.__val)
            }
            break
        
        case 4:
            {
                layer_y(local.__layerid, local.__val)
            }
            break
        
        case 7:
            {
                layer_background_htiled(local.__backid, local.__val)
            }
            break
        
        case 8:
            {
                layer_background_vtiled(local.__backid, local.__val)
            }
            break
        
        case 9:
            {
                layer_background_xscale(local.__backid, local.__val)
            }
            break
        // ...
```

**Comment by krzys-h** [comment]
I know it doesn't work on these compatibility scripts because of the way I'm injecting the type of __prop

Any other examples?

**Comment by colinator27** [comment]
I have no other examples than that right now, but I'm wondering if some other statements don't resolve correctly, like if statements with `&&` or `||` maybe? Not sure.

**Comment by colinator27** [comment]
Also, I was just testing and doing switch statements on non-temp variables doesn't seem to work, giving something like `Malformed switch statement: bad condition var (self.abc)`. (from another game, but still a potential problem)

**Comment by krzys-h** [comment]
Could you try to comment out the code that does the switch detection and show me what the orignal if statements look like? Or just decompile it with a version before the switch was implemented

**Comment by colinator27** [comment]
This is a simplification of the code in question, as an if chain:
```js
self.abc = 0
_temp_local_var_1 = self.abc
if (self.abc == -1)
    self.unrelated = "Does things";
else
{
    if (_temp_local_var_1 == 1.3) // it uses a double data type if that means anything. probably not.
    {
        self.unrelated = "Does things";
    }
    else
    {
        if (_temp_local_var_1 == 2.3)
        {
            self.unrelated = "Does things";
        }
        else
        {
            if (_temp_local_var_1 == 3.3)
            {
                self.unrelated = "Does things";
            }
            else
            {
                if (_temp_local_var_1 == 4.3)
                {
                    self.unrelated = "Does things";
                }
                else
                {
                    if (_temp_local_var_1 == 5.3)
                    {
                        self.unrelated = "Does things";
                    }
                    else
                    {
                        if (_temp_local_var_1 == 6)
                        {
                            self.unrelated = "Does things";
                        }
                        else
                        {
                            if (_temp_local_var_1 == 7)
                            {
                                self.unrelated = "Does things";
                            }
                            else
                            {
                                if (_temp_local_var_1 == 8)
                                {
                                    self.unrelated = "Does things";
                                }
                                else
                                {
                                    if (_temp_local_var_1 == 9)
                                    {
                                        self.unrelated = "Does things";
                                    }
                                    else
                                    {
                                        if (_temp_local_var_1 == 10)
                                        {
                                            self.unrelated = "Does things";
                                        }
                                        else
                                        {
                                            self.unrelated = "default";
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}
```

**Comment by krzys-h** [comment]
Actually, that double may change something because it's comparing the expressions on the left to be the same and one of them is probably `ExpressionCast(int, ExpressionVar(self.abc))` while the other is `ExpressionCast(double, ExpressionVar(self.abc))` (but the casts are hidden in the output because they made the code too unreadable and are not really relevant in high level code)

Can you try to write GML code to reproduce that?

**Comment by colinator27** [comment]
Hey, so I just fixed the problem myself. I realized that the example code I gave above actually wasn't descriptive enough! The issue was that the switch statement had `default` code that began with an if statement... Here's a pull request with my proposed fix: #63
